add secret validation option

benfleis opened the pull request:
First step in allowing users to determine validations required for `CREATE SECRET`, in relation to duckdb/duckdb-aws#108 

This PR:
- adds option to settings.json as string (not enum, not currently validated)
- fix gitignore that was ignoring settings.json!

Currently validation will occur only in the specific provider (initially AWS), which will be the immediate follow up PR.

benfleis committed:
add secret_validation setting in {'none', 'exists'}

validate string upon receipt, add matching tests

samansmink reviewed (commented):
Thanks @benfleis looks good!

I added a few small comments

benfleis commented on src/include/duckdb/main/client_config.hpp:
```diff
@@ -115,6 +115,9 @@ struct ClientConfig {
 	string http_logging_output;
 
+	//! Specify required validations for SECRET CREATE to complete successfully
+	string secret_validation;
```
The generator generated code that expected its existence - I do not claim to know what it's doing in there (yet)!

I'll fix the scope miss above, and make a custom class to validate and see where we land.

benfleis committed:
cleanup now-unneeded symbol

benfleis committed:
secret_validation to test_reset, make format-fix

benfleis committed:
fix second part of reset test

benfleis commented:
> Hi, would not have sense to the settings at the secret level, so each provider can decide what's the default (say validate, don't validate, irrelevant) ? I can see the value in having a default provided via global setting, but I would say that reading `SET secret_validation = 'exists'; CREATE SECRET s3 (.. .. );` would also carry expectation that that each secret it's being validated, while currently this depends on the secret provider/config (will be for aws, might make not sense or not being yet implemented for others).

short-answer -- there's nuance to this which I haven't documented here so far. My sense is that most user/cases either want fast-fail across the board, (or not), where a setting gives best leverage.

Taking this route enables 2 things immediately -- (1) allowing the users who relied on previous behavior to carry on cleanly and predictably, and (2) this approach gives us room to define more than 2 requirements, e.g. 'none', 'exists', 'valid' which we target as a later follow up. Additionally we discussed the possibility of a per-secret setting that would override the similar setting-based. The inconsistency in application (eg. AWS greedy fetching vs Azure lazy) will get doc'd now, and marked for future work.

Mytherin commented:
Maybe I'm missing a bit of context but it seems like this is really only applicable to `credential_chain` at this point, right?

I can't really see a use case where we would want to create an invalid secret otherwise. `credential_chain` is special because credentials are fetched externally and can thus be changed while the program is running.

Having the global setting also seems a bit confusing to me since this needs to be implemented per-secret, i.e. currently this setting has no effect at all, and presumably we will then implement this for `credential_chain` going forward, but are there plans beyond that?

If this is really just for credential chain I agree with Carlo here that adding an option to the secret (`VALIDATE false`) would be cleaner imo.

benfleis committed:
return install_s3_test_server

was removed in b755096abfadcdcec7d682ff0dfd14793d66cdb0
but appears needed by CI

benfleis committed:
Revert "return install_s3_test_server"

This reverts commit cec7d3c76a2cd78f95fad64a1104763908ce6256.

incorrect branch, revert this before closing the PR
